src/DynamicNLPModels.jl

```diff
@@ -29,11 +33,130 @@ function LQDynData(
     ul = (similar(x0,nu) .= -Inf),
     uu = (similar(x0,nu) .=  Inf)
     )
-    # TODO
+
 end
 
 function QPData(dnlp::LQDynData{VT,MT}) where {VT,MT}
-    # TODO
+    
+end
+
+""" 
+Build the (sparse) H matrix for quadratic models from Q and R matrices 
+Objective function is 1/2 z^T H z = 1/2 sum(x^T Q x for i in 1:T) + 1/2 sum(u^T R u for i in 1:(T-1))
+"""
+function build_H(
+    Q, R, nt;
+    Qf = [])
+    if size(Qf)[1] == 0
+        Qf = copy(Q)
+    end
+
+    ns = size(Q)[1]
+    nr = size(R)[1]
+
+    H = SparseArrays.sparse([],[],Float64[],(ns*nt + nr*(nt)), (ns*nt + nr*(nt)))
+
+    for i in 1:(nt-1)
+        for j in 1:ns
+            for k in 1:ns
+                row_index = (i-1)*ns + k
+                col_index = (i-1)*ns + j
+                H[row_index, col_index] = Q[k,j]
+
+            end
+        end
+    end
+
+    for j in 1:ns
+        for k in 1:ns
+            row_index = (nt-1)*ns + k
+            col_index = (nt-1)*ns + j
+            H[row_index, col_index] = Qf[k,j]
+        end
+    end
+
+
+    for i in 1:(nt-1)
+        for j in 1:nr
+            for k in 1:nr
+                row_index = ns*nt + (i-1) * nr + k
+                col_index = ns*nt + (i-1) * nr + j
+                H[row_index, col_index] = R[k,j]
+            end
+        end
+    end
+
+    return H
+end
+
+
+"""
+Build the (sparse) A matrix for quadratic models from the Ac and B matrices
+where 0 <= Az <= 0 for x_t+1 = Ac* x_t + B* u_t
+"""
+
+function build_A(Ac,B, nt)
```

Comment:
Using `A` for jacobian is confusing. Can you use `J` for jacobian and `A` for the system matrix?